File: src/main/java/no/nav/syfo/domain/dto/Sporsmal.java
package no.nav.syfo.domain.dto;

import lombok.Builder;
import lombok.Value;

import java.util.List;

@Value
@Builder
public class Sporsmal {

    String id;
    String uuid;
    String tag;
    String sporsmalstekst;
    String undertekst;
    Svartype svartype;
    String min;
    String max;
    Visningskriterie kriterieForVisningAvUndersporsmal;
    List<Svar> svar;
    List<Sporsmal> undersporsmal;
}

File: src/main/kotlin/no/nav/syfo/domain/dto/Sporsmal.kt
package no.nav.syfo.domain.dto

data class Sporsmal(

    val id: String,
    val tag: String,
    val sporsmalstekst: String? = null,
    val undertekst: String? = null,
    val svartype: Svartype? = null,
    val min: String? = null,
    val max: String? = null,
    val kriterieForVisningAvUndersporsmal: Visningskriterie? = null,
    val svar: List<Svar>? = null,
    val undersporsmal: List<Sporsmal>? = null
)
